### PR TITLE
Redesign Contributors

### LINKED ISSUE #344: Redesign: Contributors Screen
Contributors and Translators page is one of the last ones that need redesign. 

I think we may get rid of the special List Fragment for it and just use the Fragment that is used for GDEs. 
We may also rename the `GdeListFragment` to `PeopleListFragment`.

### LINKED ISSUE #424: Memory Leak in GDE adapter
Here is the leak that I got from Leak Canary. When the device was online, we were getting gde image one by one. Now it is cached for 2 days and always gotten from the cache. 
But still it leaks memory. It is really bad to do asynctask operation on getView function of an adapter. 

In org.gdg.frisbee.android.debug:2.1.0-debug:21001.
* org.gdg.frisbee.android.gde.GdeListFragment has leaked:
* GC ROOT thread java.lang.Thread.<Java Local> (named 'AsyncTask #1')
* references org.gdg.frisbee.android.gde.GdeAdapter$2.this$0 (anonymous class implements org.gdg.frisbee.android.task.CommonAsyncTask$OnBackgroundExecuteListener)
* references org.gdg.frisbee.android.gde.GdeAdapter.mDataSetObservable
* references android.database.DataSetObservable.mObservers
* references java.util.ArrayList.array
* references array java.lang.Object[].[0]
* references org.gdg.frisbee.android.common.GdgListFragment$3.this$0 (anonymous class extends android.database.DataSetObserver)
* leaks org.gdg.frisbee.android.gde.GdeListFragment instance

* Reference Key: ed0decfc-1521-4ab3-8970-fb134c134ad7
* Device: motorola google Nexus 6 shamu
* Android Version: 5.1.1 API: 22 LeakCanary: 1.3.1
* Durations: watch=5006ms, gc=176ms, heap dump=5461ms, analysis=22399ms

* Details:
* Instance of java.lang.Thread
|   static $staticOverhead = byte[] [id=0x7110acc9;length=48;size=64]
|   static MAX_PRIORITY = 10
|   static MIN_PRIORITY = 1
|   static NANOS_PER_MILLI = 1000000
|   static NORM_PRIORITY = 5
|   static count = 22217
|   static defaultUncaughtHandler = com.android.internal.os.RuntimeInit$UncaughtHandler [id=0x12c096c0]
|   contextClassLoader = dalvik.system.PathClassLoader [id=0x12c6dfe0]
|   daemon = false
|   group = java.lang.ThreadGroup [id=0x70fa6148]
|   hasBeenStarted = true
|   id = 22173
|   inheritableValues = null
|   interruptActions = java.util.ArrayList [id=0x12da7b60]
|   localValues = java.lang.ThreadLocal$Values [id=0x12db02a0]
|   lock = java.lang.Object [id=0x12d73ec0]
|   name = java.lang.String [id=0x12da7b40]
|   nativePeer = -1264360448
|   parkBlocker = null
|   parkState = 1
|   priority = 5
|   stackSize = 0
|   target = java.util.concurrent.ThreadPoolExecutor$Worker [id=0x12d71e80]
|   uncaughtHandler = null
* Instance of org.gdg.frisbee.android.gde.GdeAdapter$2
|   this$0 = org.gdg.frisbee.android.gde.GdeAdapter [id=0x12edc160]
* Instance of org.gdg.frisbee.android.gde.GdeAdapter
|   mClient = com.google.api.services.plus.Plus [id=0x12edc250]
|   mConsumedMap = java.util.HashMap [id=0x12edc190]
|   mContext = org.gdg.frisbee.android.gde.GdeActivity [id=0x1356ff80]
|   mGdes = java.util.ArrayList [id=0x12ecab20]
|   mInflater = com.android.internal.policy.impl.PhoneLayoutInflater [id=0x12dce640]
|   mJsonFactory = com.google.api.client.json.gson.GsonFactory [id=0x12db1ef0]
|   mPlusPattern = java.util.regex.Pattern [id=0x12ecab40]
|   mTransport = org.gdg.frisbee.android.api.GapiOkTransport [id=0x12db1ee0]
|   mDataSetObservable = android.database.DataSetObservable [id=0x12db1ed0]
* Instance of android.database.DataSetObservable
|   mObservers = java.util.ArrayList [id=0x12ecab00]
* Instance of java.util.ArrayList
|   static $staticOverhead = byte[] [id=0x70ff1301;length=16;size=32]
|   static MIN_CAPACITY_INCREMENT = 12
|   static serialVersionUID = 8683452581122892189
|   array = java.lang.Object[] [id=0x12e2c780;length=12]
|   size = 1
|   modCount = 3
* Array of java.lang.Object[]
|   [0] = org.gdg.frisbee.android.common.GdgListFragment$3 [id=0x12db1fe0]
|   [1] = null
|   [2] = null
|   [3] = null
|   [4] = null
|   [5] = null
|   [6] = null
|   [7] = null
|   [8] = null
|   [9] = null
|   [10] = null
|   [11] = null
* Instance of org.gdg.frisbee.android.common.GdgListFragment$3
|   this$0 = org.gdg.frisbee.android.gde.GdeListFragment [id=0x12c51500]
* Instance of org.gdg.frisbee.android.gde.GdeListFragment
|   static $staticOverhead = byte[] [id=0x13716001;length=16;size=32]
|   static ARG_ACTIVE = java.lang.String [id=0x1312bd00]
|   static ARG_GDES = java.lang.String [id=0x1312bd80]
|   mAdapter = org.gdg.frisbee.android.gde.GdeAdapter [id=0x12edc160]
|   mAdapter = org.gdg.frisbee.android.gde.GdeAdapter [id=0x12edc160]
|   mEmptyText = null
|   mEmptyView = null
|   mHandler = android.os.Handler [id=0x12eb53a0]
|   mList = null
|   mListShown = false
|   mLoading = false
|   mOnClickListener = org.gdg.frisbee.android.common.GdgListFragment$2 [id=0x12db1c50]
|   mProgressContainer = null
|   mRequestFocus = org.gdg.frisbee.android.common.GdgListFragment$1 [id=0x12db1c40]
|   mActivity = null
|   mAdded = false
|   mAllowEnterTransitionOverlap = null
|   mAllowReturnTransitionOverlap = null
|   mAnimatingAway = null
|   mArguments = android.os.Bundle [id=0x12eb53e0]
|   mBackStackNesting = 0
|   mCalled = true
|   mCheckedForLoaderManager = false
|   mChildFragmentManager = null
|   mContainer = null
|   mContainerId = 0
|   mDeferStart = false
|   mDetached = false
|   mEnterTransition = null
|   mEnterTransitionCallback = null
|   mExitTransition = null
|   mExitTransitionCallback = null
|   mFragmentId = 0
|   mFragmentManager = null
|   mFromLayout = false
|   mHasMenu = false
|   mHidden = false
|   mInLayout = false
|   mIndex = -1
|   mInnerView = null
|   mLoaderManager = null
|   mLoadersStarted = false
|   mMenuVisible = false
|   mNextAnim = 0
|   mParentFragment = null
|   mReenterTransition = java.lang.Object [id=0x133add30]
|   mRemoving = false
|   mRestored = false
|   mResumed = false
|   mRetainInstance = false
|   mRetaining = false
|   mReturnTransition = java.lang.Object [id=0x133add30]
|   mSavedFragmentState = null
|   mSavedViewState = android.util.SparseArray [id=0x130733c0]
|   mSharedElementEnterTransition = null
|   mSharedElementReturnTransition = java.lang.Object [id=0x133add30]
|   mState = 0
|   mStateAfterAnimating = 0
|   mTag = null
|   mTarget = null
|   mTargetIndex = -1
|   mTargetRequestCode = 0
|   mUserVisibleHint = false
|   mView = null
|   mWho = null

### PR DESCRIPTION
**Pull request by tasomaniac**
Fixes #344 and hopefully fixes #424 

This work contains #422 
I had to work on top of that branch. The changes here includes the changes from #422. So if we can review and merge #422 first. This can be reviewed a little bit easily. 

This removes all the adapters and xml item layouts for Contributor and Translator screens. 
GdeFragment is refactored to more generic class which is PeopleListFragment. What it does is to display list of people. It doesn't matter if the person is a GDE, Contributor or Translator. 

And finally AsyncTask inside GDE Adapter is removed. It is really simple now. Instead we have `PlusPersonDownloader` now which automatically first gets the Person object from Google plus and returns the image URL to Picasso. Since Picasso caches based on URL addresses, it will directly displays the image without requesting ModelCache, Plus API or anything. And lastly it will hopefully remove the memory leak described in #424 

Here are the screenshots: 
screen shot 2015-07-27 at 19 59 50
screen shot 2015-07-27 at 19 59 46

### CODE REVIEW

**Comment by PareshMayani** [comment]
Looks perfect to me!

**Comment by tasomaniac** [comment]
Thank you @PareshMayani 
Since this is a big PR, we may wait for 2nd person to review it.

File: app/src/main/java/org/gdg/frisbee/android/app/App.java [hunk 1]
```diff
@@ -150,11 +153,17 @@ public void onCreate() {
         PrefUtils.increaseAppStartCount(this);
 
         // Initialize Picasso
+        // When we clone mOkHttpClient, it will use all the same cache and everything.
+        // Only the interceptors will be different.
+        // We shouldn't have the below interceptor in other instances.
+        OkHttpClient picassoClient = mOkHttpClient.clone();
```
Comment (fnk0) [inline app/src/main/java/org/gdg/frisbee/android/app/App.java]: May I ask why you decides to clone the OkhttpClient rather than just use the current one?

**Comment by fnk0** [comment]
It looks good to me. I like the refactor of the different types of people fragment and how they are all treated the same now (makes maintenance easier). 

I have some opinions of things I think it should change but those should be separate issues for future PR's (such as changing ListView to RecyclerView).

File: app/src/main/java/org/gdg/frisbee/android/app/App.java (hunk 1 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/app/App.java]: Picasso client uses an interceptor for plus profile images. When you clone it, it uses the same instance of the internal objects in it. Performance wise it is the same. I found that is Jake Wharton s recommendation for interceptors

Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/app/App.java]: More info about this: 
https://github.com/square/picasso/issues/900#issuecomment-75133184

**Comment by tasomaniac** [comment]
I am merging this since 2 people reviewed it.

**Pull request merged**